pkg/server/cluster_server.go

```diff
@@ -42,23 +52,61 @@ type clusterServer struct {
 // It accepts the kubeConfig which is not required when it's
 // run from within the cluster(useful in testing).
 // It accepts the apiserverURL which is the location of the KubeAPIServer.
-func NewClusterServer(kubeConfig, apiserverURL string) (Server, error) {
+func NewClusterServer(kubeConfig, apiserverURL string, enableMaster bool) (Server, error) {
 	restConfig, err := getClientConfig(kubeConfig)
 	if err != nil {
 		return nil, fmt.Errorf("Failed to create Kubernetes rest client: %v", err)
 	}
 
+	kc, err := clientset.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
 	mc := v1.NewForConfigOrDie(restConfig)
 	return &clusterServer{
+		kubeClient:     kc,
 		machineClient:  mc,
 		kubeconfigFunc: func() ([]byte, []byte, error) { return kubeconfigFromSecret(bootstrapTokenDir, apiserverURL) },
+		enableMaster:   enableMaster,
 	}, nil
 }
 
 // GetConfig fetches the machine config(type - Ignition) from the cluster,
 // based on the pool request.
-func (cs *clusterServer) GetConfig(cr poolRequest) (*ignv2_2types.Config, error) {
-	mp, err := cs.machineClient.MachineConfigPools().Get(cr.machineConfigPool, metav1.GetOptions{})
+func (cs *clusterServer) GetConfig(cr poolRequest, auth string) (*ignv2_2types.Config, error) {	
+	authSecretObj, err := cs.kubeClient.CoreV1().Secrets(componentNamespace).Get(ignitionAuth, metav1.GetOptions{})
+	authEnabled := true
+	if err != nil {
+		// For backwards compat, allow any request if the secret isn't found
+		if apierrors.IsNotFound(err) {
+			authEnabled = false
+		} else {
+			return nil, errors.Wrapf(err, "getting ignition-auth")
+		}
+	}
+
+	name := cr.machineConfigPool
+
+	// If there's a secret, require that it was passed as a query parameter.
+	if authEnabled {
+		authSecret := string(authSecretObj.Data[name])
```

Comment:
If an administrator forgets to configure an authsecret for a new machine pool, this will fail open.

Reply:
That's intentional for backcompat reasons.